socs/agents/hwp_pid/drivers/pid_controller.py

```diff
@@ -236,17 +251,17 @@ def get_freq(self):
         responses = []
         responses.append(self.send_message("*X01"))
         decoded_resp = self.return_messages(responses)[0]
-        attempts = 3
-        for attempt in range(attempts):
-            if self.verb:
-                print(responses)
-                print(decoded_resp)
-            if decoded_resp.msg_type == 'measure':
-                return decoded_resp.measure
-            elif decoded_resp.msg_type == 'error':
-                print(f"Error reading freq: {decoded_resp.msg}")
-        raise ValueError('Could not get current frequency')
+        if self.verb:
+            print(responses)
+            print(decoded_resp)
+        if decoded_resp.msg_type == 'measure':
+            return decoded_resp.measure
+        elif decoded_resp.msg_type == 'error':
+            raise ValueError(f"Error reading freq: {decoded_resp.msg}")
```

Comment:
This will be swallowed by the `try/catch` in the `retry_multiple_times` decorator, so it won't be logged to stdout. Make sure to log an error here with the problematic response.